scripts/west_commands/runners/linkserver.py

```diff
@@ -187,8 +187,10 @@ def flash(self, **kwargs):
         if self.erase:
             self.do_erase()
 
-        # Use .hex or .bin, preferring .hex over .bin
-        if self.supports_hex and self.hex_name is not None and os.path.isfile(self.hex_name):
+        # Use .elf, .hex or .bin, preferring .elf over .hex over .bin
+        if self.elf_name is not None and os.path.isfile(self.elf_name):
+            flash_cmd = (["load", self.elf_name])
+        elif self.supports_hex and self.hex_name is not None and os.path.isfile(self.hex_name):
```

Comment:
To have the same behavior as for all other runners, could you give the highest priority to .hex ?

Reply:
Put hex back to default, thanks for the remark.

Reply:
> Put hex back to default, thanks for the remark.

Thanks.

I have reviewed all runners, and .elf is used only for debugging.
For flashing are used .hex and .bin. 
Only one OpenOCD runner is able to flash .elf, BUT only if the "use_elf" parameter is set intentionally.
Are you sure that you cannot use .hex for flashing instead?
Afraid, that can be some examples that require flashing of shifted .bin files, and flashing of .elf can brake them.

Reply:
The internal flash on a MCXN is on address `0x0000 0000` the external flash is on `0x8000 0000` when using both regions in your program results in a a generated hex file of around ~3GB. When using `west flash` it simply fails because I think it's actually trying to write the whole region.

ELF pretty much solves this because it contains a list of sections that need to be programmed.

Reply:
> The internal flash on a MCXN is on address `0x0000 0000` the external flash is on `0x8000 0000` when using both regions in your program results in a a generated hex file of around ~3GB. When using `west flash` it simply fails because I think it's actually trying to write the whole region.
> 
> ELF pretty much solves this because it contains a list of sections that need to be programmed.

Hi @PetervdPerk-NXP, 
could you point to the example, to test it?

Reply:
Added a new commit it seemed that Linkserver version detect never worked as intended thus faulting when flashing ihex files.

> ~~The internal flash on a MCXN is on address 0x0000 0000 the external flash is on 0x8000 0000 when using both regions in your program results in a a generated hex file of around 3GB. When using west flash it simply fails because I think it's actually trying to write the whole region.~~

Okay it turns you've to set `CONFIG_BUILD_NO_GAP_FILL=y` to avoid generating big hex files.

> Only one OpenOCD runner is able to flash .elf, BUT only if the "use_elf" parameter is set intentionally.

Sure it seems that the other runners haven't implemented it yet, however the API exposes `self.elf_name` and when `CONFIG_BUILD_OUTPUT_HEX=n` and `CONFIG_BUILD_OUTPUT_BIN=y` the build system only outputs an elf file. Hence I believe we should support ELF files for our runner.

Reply:
Added the DNM label.
As we have discussed off-line with @PetervdPerk-NXP
-  it will break MCUBoot examples by flashing .elf instead of signed.bin.
- the huge .hex/bin issue can be solved by 
```
CONFIG_BUILD_OUTPUT_HEX=y
CONFIG_BUILD_NO_GAP_FILL=y
CONFIG_BUILD_OUTPUT_BIN=n
```

My proposal:
The ,hex and .bin generation is optional.
As compromise, we can decrease .elf flashing priority. 
So the flashing priority can be hex>bin>elf.
As the result, no existing example is broken and the .elf flashing can be supported.